### ISSUE TITLE
Bookmarks (searches and documents)

### ISSUE THREAD

**Issue by simongray** [kuhumcst/glossematics#72]
It would be a valuable addition to allow storing searches and specific documents as bookmarks, easily accessible by the user. Obviously, this feature requires persistence, so it needs to use the same datastore as the user comments #52.

**Comment by simongray** [comment]
(I think I will start with the bookmarks feature as this is simpler implementation-wise compared with #52)

A bookmark is an entity where the following information is deemed relevant:

```clojure
{:bookmark/ident    ident
 :bookmark/path     path
 :entity/visibility visibility  ; e.g. :public
 :bookmark/added    date}
```

Bookmarks need to be persisted, i.e. we will probably need to combine a persisted graph with the in-memory one using [asami wrapgraph](https://github.com/quoll/asami/blob/main/src/asami/wrapgraph.cljc). The persisted graph should exist in a Docker volume, probably somewhere on disk that gets backed up by KU-IT.

In time, it may be beneficial to view the bookmarks of other users or dynamically compile a list of popular bookmarks. The `:entity/visibility` attribute is intended to reflect this desire while pre-empting any possible GDPR issues, e.g. bookmarks will default to being publicly visible.

**Comment by simongray** [comment]
For this purpose, it will be optimal to query/transact to the separate persisted graph directly, since it only interfaces with paths rather than actual document data.

**Comment by simongray** [comment]
Nice-to-have stuff that is (as of now) still TODO:

* Frontend changes should take effect immediately and use the callback from the backend request to correct the frontend changes, i.e. clicking on the bookmark button should add it in the frontend and wait for a potentially unsuccessful backend request to remove the bookmark.

**Comment by simongray** [comment]
Optimistic frontend changes implemented in 242172e440a5d0ce38993051a05cde10d10e260f

**Comment by simongray** [comment]
Another nice-to-have documented in https://github.com/kuhumcst/glossematics/issues/83
